UC NetCode Fall 2024/Assets/App/Resource/Scripts/NetHelpers/HealthNetScript.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using Unity.Netcode;
+
+namespace App.Resource.Scripts.Player
+{
+    public class HealthNetScript : NetworkBehaviour
+    {
+        [SerializeField] public float _startingHealth = 100f;
+        [SerializeField] public float cooldown = 1.5f;
+        [SerializeField] private bool _canDamage = true;
+        [SerializeField] private NetworkVariable<float> _Health = new NetworkVariable<float>(100);
+        public Image _healthbar;
+
+        private GameScript _gamescript;
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+
+            _Health.Value = _startingHealth;
+
+            _Health.OnValueChanged += UpdateHealth;
```

Comment:
We need to tell the gameScript we are an active player, by calling `FindObjectOfType<GameScript>().AddPlayerRpc();`  here in the OnNetworkSpawn override.